File: 020-valid-parentheses/solution.go
func isValid(s string) bool {
    brackets := map[string]string{
        ")": "(",
        "]": "[",
        "}": "{",
    }
    var visit []string
    for i := 0; i < len(s); i++ {
        char := string(s[i])
        if _, ok := brackets[char]; ok {
            if len(visit) > 0 {
                top := visit[len(visit)-1]
                if top == brackets[char] {
                    visit = visit[:len(visit)-1]
                } else {
                    return false
                }
            } else {
                return false
            }
        } else {
            visit = append(visit, char)
        }
    }
    return len(visit) == 0
}

File: 020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        brackets = {
            ')':'(',
            '}':'{',
            ']':'['
        }
        visit = []
        str = list(s)

        for c in str:
            if c not in brackets:
                visit.append(c)
            elif c in brackets:
                if len(visit) != 0:
                    removed_ele = visit[-1]
                    if removed_ele == brackets[c]:
                        visit.pop()
                        continue
                    else:
                        break
                else:
                    visit.append(c)
                    
        return len(visit)==0
        